src/org/parosproxy/paros/core/scanner/AbstractPlugin.java

```diff
@@ -272,6 +269,11 @@ protected void sendAndReceive(HttpMessage message, boolean isFollowRedirect, boo
                 }
             }
         }
+        
+        //ZAP: Set the ZAP scan id header here instead at the init phase 

```

Comment:
We should not add ZAP comments in the code, they should be after the license header (in this case seems unnecessary the existing one already covers it).

Reply:
@thc202 

This is noted. I just updated the pull request.